src/compiler/types.ts

```diff
@@ -1511,6 +1517,32 @@ namespace ts {
         typeExpression: JSDocTypeExpression;
     }
 
+    // @kind(SyntaxKind.JSDocTypedefTag)
+    export interface JSDocTypedefTag extends JSDocTag, Declaration, JSDocParentTag {
+        name: Identifier;
+        typeExpression?: JSDocTypeExpression;
+        type: JSDocType;
+    }
+
+    export interface JSDocParentTag extends JSDocTag {
+        jsDocPropertyTags?: NodeArray<JSDocPropertyTag>;
+        jsDocTypeTag?: JSDocTypeTag;
+    }
+
+    // @kind(SyntaxKind.JSDocPropertyTag)
+    export interface JSDocPropertyTag extends JSDocTag, TypeElement {
+        name: Identifier;
+        typeExpression: JSDocTypeExpression;
+        // Add a "type" property here to make the interface compatible 
+        // with the "VariableLikeDeclaration" definition

```

Comment:
Because the `@property` tag is essential the same as a property declaration, so all operations we supported for property declaration should support `@property` tag automatically. In the compiler many operations on property declaration are treated abstractly as `VariableLikeDeclaration`; making it share the same shape avoids treating `@property` differently in many places. For example [here](https://github.com/Microsoft/TypeScript/blob/master/src/compiler/checker.ts#L2808)